packages/core/services/FileService/DatabaseFileService/index.ts

```diff
@@ -130,55 +130,95 @@ export default class DatabaseFileService implements FileService {
             .select(annotations.map((annotation) => `"${annotation}"`).join(", "))
             .from(this.dataSourceNames);
 
-        selections.forEach((selection) => {
-            selection.indexRanges.forEach((indexRange) => {
+        this.applySelectionFilters(sqlBuilder, selections);
+
+        return this.handleFileDownload(sqlBuilder.toSQL(), format);
+    }
+
+    /**
+     * Processes selections and applies WHERE clause directly to the SQLBuilder.
+     */
+    private applySelectionFilters(sqlBuilder: SQLBuilder, selections: Selection[]): void {
```

Comment:
This looks like it should be a static method. Moreover, it feels like this belongs in the `SQLBuilder` domain since it is scoped as a setter-like side-effect function

Reply:
moved to static in [94e06d3](https://github.com/AllenInstitute/biofile-finder/pull/437/commits/94e06d3c39d9c7506c6094335070803b6b8d6a66)